torch/utils/data/dataloader.py

```diff
@@ -268,17 +308,25 @@ class DataLoader(object):
             if the dataset size is not divisible by the batch size. If False and
             the size of dataset is not divisible by the batch size, then the last batch
             will be smaller. (default: False)
+        timeout (int, optional): if positive, the timeout value for collecting a batch
+            from workers. (default: 0)
     """
 
     def __init__(self, dataset, batch_size=1, shuffle=False, sampler=None, batch_sampler=None,
-                 num_workers=0, collate_fn=default_collate, pin_memory=False, drop_last=False):
+                 num_workers=0, collate_fn=default_collate, pin_memory=False, drop_last=False,
+                 timeout=0):
         self.dataset = dataset
         self.batch_size = batch_size
         self.num_workers = num_workers
         self.collate_fn = collate_fn
         self.pin_memory = pin_memory
         self.drop_last = drop_last
 
+        if not isinstance(timeout, int):
+            raise ValueError('timeout must be an integer')
```

Comment:
I don't think this is necessary. We usually don't do type checks, and it would be fine for the timeout to be a float or long. Just check that it's `>= 0` if you want

Reply:
I forgot to remove this :) It was needed when using SIGALRM. Thanks for pointing out.